superset-frontend/src/datasource/DatasourceEditor.jsx

```diff
@@ -58,6 +58,10 @@ const DatasourceContainer = styled.div`
   .change-warning .bold {
     font-weight: ${({ theme }) => theme.typography.weights.bold};
   }
+
+  .help-block {
+    margin-top: 8px;
+  }
```

Comment:
when I render this, I still see the margin-bottom of the form-group, so it's still making a total of 16px. 
This is a bit tricky actually. I read this as

form-group should always have a margin bottom of 8px
when there are two siblings (form-group followed by a form-group), then the second one should have a margin-top of 16px. 

I think if you go that way, you won't need to add anything additional to the help-block. 
lmk if that helps

Reply:
Yeah that helps a lot, it really cleared up my thinking about it. I also ended up reading more about how margin works. I ended up trying something else, because nothing I was reading was helping me find a way of implementing your solution. 

I ended up targeting the specific form group that is above each help block (form-group-md). I changed the margin-bottom of these to be 8px. However, in order to make sure that 'Extra' and 'Owners' section had a help block with a margin of 8px, I also added a margin-top to the help-block. 

Originally I thought that a margin-bottom of 8 and a margin-top of 8 on two elements would make 16px, but they don't actually add the way that I thought they might. It simple goes to the highest margin value. Which is cool. 

It looks like this now:
![image](https://user-images.githubusercontent.com/48933336/107062308-2a313780-67a7-11eb-960a-984077256bfd.png)

The margin between form groups is 16px:
![image](https://user-images.githubusercontent.com/48933336/107062487-5a78d600-67a7-11eb-9c1b-bf6f8f90bdbe.png)

The margin between help-block and form-group is 8px:
![image](https://user-images.githubusercontent.com/48933336/107062563-6fee0000-67a7-11eb-8820-87cee474827e.png)

Reply:
This look great. One other thing I was hinting at was sibling selectors: `when there are two siblings (form-group followed by a form-group), then the second one should have a margin-top of 16px.` So that would be `.form-group + .form-group` but it looks like you made it work this way, too.